src/executeQuery.js

```diff
@@ -76,9 +79,14 @@ function esriFy (result, options, i) {
     })
   }
 
-  const metadata = (options.collection && options.collection.metadata) || {}
-  if (!metadata.idField) {
-    result.attributes.OBJECTID = i
+  // If the idField for the model set and is not already called OBJECTID, use its value as OBJECTID
+  if (idField && idField !== 'OBJECTID') {
+    result.attributes.OBJECTID = result.attributes[idField]
+    delete result.attributes[idField]
```

Comment:
I'm confused, are we remapping the name of any type objectid field to be called OBJECTID? if so why?

Reply:
If there is an `idField` set in provider metadata, it is assumed we want to use that for the value of OBJECTID.  If we don't then delete `result.attributes[idField]`  this data would show up in the payload twice: once in `result.attributes.OBJECTID` and once in `result.attributes[idField]`.  Thus, the `delete`.  However, this was an uncertainty for me - perhaps we want to maintain both fields?

The `delete` causes problems if the provider metadata implementation has an `idField: 'OBJECTID'` (as was done in test fixtures).  The `delete` on line 85 would then be equivalent to `delete result.attributes.OBJECTID`, and without an OBJECTID attribute, ArcGIS clients fail.  Thus the additional check `idField !== 'OBJECTID'`

Reply:
> If there is an idField set in provider metadata, it is assumed we want to use that for the value of OBJECTID.

That's not correct. We just want to set the metadata of the layer to specify that the field passed in is the objectid. It does not have to be named `OBJECTID` and doing so causes you a bunch of juggling later which is unecessary.

Reply:
My user testing in Pro suggests that I need to have an attribute with key 'OBJECTID' as well as a field object with `name` and `alias` values of 'OBJECTID'.  However, there are many permutations of those assignments and perhaps I have mis-interpreted the Pro requirement from user-testing results(?)

Or perhaps you are suggesting that we add an OBJECTID attribute and also maintain the other attribute referenced by `idField`?

Reply:
> My user testing in Pro suggests that I need to have an attribute with key 'OBJECTID' as well as a field object with name and alias values of 'OBJECTID'.

I'm suggesting that perhaps we are not properly setting the objectid field. There are two places for this metadata. In the fields array and as a direct key on the layer json. I'm about 99.99% sure OBJECTID fields don't have be be named this.

> Or perhaps you are suggesting that we add an OBJECTID attribute and also maintain the other attribute referenced by idField?

I'd say yes if I'm wrong about the above.

Reply:
Ok, yes I see what you mean.  FeatureServer is not using the provider metadata to construct the `fields` array for the layer service.  I'll start there.

Reply:
**TL;DR;** Tried setting metadata in layer and query services differently, but Pro crashes.

I've gone back to FeatureServer and in a branch, and in the layer and query services I have tried the creating a field object of type `esriFieldTypeOID` with an name and alias of value determine by the metadata's `idField`. For example:
```
        {
            "name": "featureId",
            "type": "esriFieldTypeOID",
            "alias": "featureId",
            "sqlType": "sqlTypeOther",
            "domain": null,
            "defaultValue": null,
            "editable": false,
            "nullable": false
        }
```
I've made sure that the layer service also has `"objectIdField": "featureId"` and that the query service has `"objectIdFieldName": "featureId"`.

Also tried keeping the `alias` set to 'OBJECTID', but result was same, Pro Crashes.

Reply:
it crashes!?!?!??! you gotta send that bug report along. I guess we'll fall back to duplicating the field as it's original name + OBJECTID. At least that way both can be filtered on.

Reply:
The duplication gets a little weird if `idField: 'OBJECTID`.  If we always, duplicate, that situation would create two field objects named 'OBJECTID', so we'll need to manage that case.  Thanks to the test-fixtures for exposing that.

Reply:
Yep, only duplicate if the idField is not called OBJECTID. Make sure to leave some copious notes in the code as to why we're going this route btw.

Reply:
Support for duplication in metadata `fields` arrays [here](https://github.com/koopjs/FeatureServer/pull/96).  This approach will change what's needed in winnow, so closing this PR for now.